src/ops/convert.rs

```diff
@@ -8,14 +8,27 @@ fn cast(pool: &TensorPool, input: Input, dtype: DataType) -> Result<Output, OpEr
         DataType::Int32 => match input {
             Input::Int32Tensor(t) => Ok(t.map_in(pool, |x| *x).into()),
             Input::FloatTensor(t) => Ok(t.map_in(pool, |x| *x as i32).into()),
-            _ => Err(OpError::UnsupportedType),
+            Input::Int8Tensor(t) => Ok(t.map_in(pool, |x| *x as i32).into()),
+            Input::UInt8Tensor(t) => Ok(t.map_in(pool, |x| *x as i32).into()),
         },
         DataType::Float => match input {
             Input::FloatTensor(t) => Ok(t.map_in(pool, |x| *x).into()),
             Input::Int32Tensor(t) => Ok(t.map_in(pool, |x| *x as f32).into()),
-            _ => Err(OpError::UnsupportedType),
+            Input::Int8Tensor(t) => Ok(t.map_in(pool, |x| *x as f32).into()),
+            Input::UInt8Tensor(t) => Ok(t.map_in(pool, |x| *x as f32).into()),
+        },
+        DataType::Int8 => match input {
```

Comment:
I think that using `as` for casts is fine for all cases here. Just making some notes for myself:

Converting non-i8 types to i8 will truncate in various ways, and the same for u8. The `Cast` op specs are [here](https://onnx.ai/onnx/operators/onnx__Cast.html). The main cases of interest are:

- Float -> int when out of range: ONNX spec says undefined, so an `as` cast is fine
- Int -> Int when out of range: ONNX spec says `when OOR, discard higher bits and reinterpret (with respect to two’s complement representation for signed types). For example, 200 (int16) -> -56 (int8).`. This matches the behavior of `as` in Rust